### PR TITLE
Default to spectral variant on simplices

### PR DESCRIPTION
**Pull request by pbrubeck**
Depends on https://github.com/FInAT/FInAT/pull/112
fixes https://github.com/firedrakeproject/firedrake/issues/3154

### CODE REVIEW

**Review by JDBetteridge — CHANGES_REQUESTED** [review]
Can you open a draft PR on firedrake with the do not merge label, which tests this branch, please. Also resolve the conflicts.

File: tests/test_create_fiat_element.py [hunk 1]
```diff
@@ -101,16 +101,14 @@ def test_interval_variant(family, variant, expected_cls):
     assert isinstance(create_element(ufl_element), expected_cls)
 
 
-def test_triangle_variant_spectral_fail():
-    ufl_element = finat.ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
-    with pytest.raises(ValueError):
-        create_element(ufl_element)
+def test_triangle_variant_spectral():
+    ufl_element = ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
```
Comment (pbrubeck) [inline tests/test_create_fiat_element.py:105]:
```suggestion
    ufl_element = finat.ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
```

File: tests/test_create_finat_element.py [hunk 2]
```diff
@@ -90,16 +90,14 @@ def test_interval_variant(family, variant, expected_cls):
     assert isinstance(create_element(ufl_element), expected_cls)
 
 
-def test_triangle_variant_spectral_fail():
-    ufl_element = finat.ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
-    with pytest.raises(ValueError):
-        create_element(ufl_element)
+def test_triangle_variant_spectral():
+    ufl_element = ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
```
Comment (pbrubeck) [inline tests/test_create_finat_element.py:94]:
```suggestion
    ufl_element = finat.ufl.FiniteElement('DP', ufl.triangle, 2, variant='spectral')
```

File: tests/test_create_finat_element.py [hunk 3]
```diff
@@ -90,16 +90,14 @@ def test_interval_variant(family, variant, expected_cls):
...
+    create_element(ufl_element)
 
 
-def test_triangle_variant_spectral_fail_l2():
-    ufl_element = finat.ufl.FiniteElement('DP L2', ufl.triangle, 2, variant='spectral')
-    with pytest.raises(ValueError):
-        create_element(ufl_element)
+def test_triangle_variant_spectral_l2():
+    ufl_element = ufl.FiniteElement('DP L2', ufl.triangle, 2, variant='spectral')
```
Comment (pbrubeck) [inline tests/test_create_finat_element.py:99]:
```suggestion
    ufl_element = finat.ufl.FiniteElement('DP L2', ufl.triangle, 2, variant='spectral')
```

**Review by connorjward — APPROVED** [review]
This is passing Firedrake tests [here](https://github.com/firedrakeproject/firedrake/pull/3193). In future @pbrubeck please link between that PR and this one so it can be found.

**Pull request merged**